src/Java.Interop.Tools.JavaCallableWrappers/Java.Interop.Tools.JavaCallableWrappers/TypeNameMapGenerator.cs

```diff
@@ -202,6 +202,15 @@ static void WriteBinaryMapping (Stream o, Dictionary<string, string> mapping)
 			o.WriteByte (0x0);
 		}
 
+		/// <summary>
+		/// We need the equivalent of GetAssemblyQualifiedName, but without the Assembly's FullName
+		/// NOTE: Cecil also uses / instead of + for nested types, see Java.Interop.Tools.Cecil.TypeDefinitionRocks.GetAssemblyQualifiedName
```

Comment:
I have to wonder: should `TypeDefinitionRocks.GetAssemblyQualifiedName()` be updated to do this instead of having it emit "proper" assembly-qualified names? (If we did this, we'd also want to rename it to e.g. `.GetAbbreviatedAssemblyQualifiedName()`?)

Looking at the callers of `TypeDefinitionRocks.GetAssemblyQualifiedName()`, a worrying one is in [`JavaCallableWrapperGenerator.cs`](https://github.com/xamarin/java.interop/blob/57d5d514c41cba1af3e1dc71ea157cff0c5a43b6/src/Java.Interop.Tools.JavaCallableWrappers/Java.Interop.Tools.JavaCallableWrappers/JavaCallableWrapperGenerator.cs#L570), for the static constructor, e.g. [from our unit tests](https://github.com/xamarin/java.interop/blob/master/src/Java.Interop.Tools.JavaCallableWrappers/Test/Java.Interop.Tools.JavaCallableWrappers/JavaCallableWrapperGeneratorTests.cs#L166):

```java
	mono.android.Runtime.register (
		"Xamarin.Android.ToolsTests.ExportsMembers, Java.Interop.Tools.JavaCallableWrappers-Tests, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null",
		ExportsMembers.class,
		__md_methods);
```

*All* "full" Assembly-qualified name instances need to be updated, including (*especially* including) the Java Callable Wrappers, and this PR doesn't touch them.

Reply:
Ah I just found `TypeDefinitionRocks.GetPartialAssemblyQualifiedName`, which just needs a minor fix and we should use it.

Reply:
What minor fix does it need?

`TypeDefinitionRocks.GetPartialAssemblyQualifiedName()` is used in xamarin-android, in [`src/Xamarin.Android.Build.Tasks/Tasks/GenerateJavaStubs.cs`](https://github.com/xamarin/xamarin-android/blob/cf185ad3b7234b714ffd16c19c18a76cb07561d4/src/Xamarin.Android.Build.Tasks/Tasks/GenerateJavaStubs.cs#L165), so be careful about semantics.